process/block/metablockProposal.go

```diff
@@ -522,7 +521,8 @@ func (mp *metaProcessor) checkEpochCorrectnessV3(
 	}
 
 	isEpochStartBlock := headerHandler.IsStartOfEpochBlock()
-	hasAllEpochStartData := hasEpochStartExecutionResults && isEpochStartBlock && wasEpochStartProposed
+	epochStartDataMatches := mp.epochStartData.Equal(headerHandler.GetEpochStartHandler())
+	hasAllEpochStartData := hasEpochStartExecutionResults && isEpochStartBlock && wasEpochStartProposed && epochStartDataMatches
 	hasAnyEpochStartData := hasEpochStartExecutionResults || isEpochStartBlock || wasEpochStartProposed
 	hasIncompleteEpochStartData := hasAnyEpochStartData && !hasAllEpochStartData
```

Comment:
Would it make sense to add this `|| epochStartDataMatches` in `hasAnyEpochStartData`?